fecgov/fec-cms
# Feature/4187 list child nodes

## Summary

- Resolves #4187 

There were a few places across the site where we had lists whose children were not list-item elements. [Rules](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) are that `<ol>` and `<ul>` can only have child nodes that are `<li>`, `<script>`, or `<template>`. Lighthouse was flagging this for us as an accessibility issue.

These were the instances I found but there could be others.

## Impacted areas of the application

**Homepage**: The News and announcements section was wrapped in a `<ul>` instead of a `<div>`. That's fixed.

**Terms**: There was one glossary definition where we were using three `<ul>` with bare text and an `<a>` directly inside. They've been updated, using another term on [another page](https://www.fec.gov/help-candidates-and-committees/making-disbursements-political-party/definition-federal-election-activity-fea/) (click "FEA") as a reference.

**Advisory opinions**:
- we were using a `<ul>` as a wrapper for `<div>`s
- we were opening a `<ul>` inside a conditional but closing it outside of it, the result being a random `</ul>` when there were no results to show
- adjusted html indentation down the document
- adjusted logic indentation for consistency and to more clearly show the `for` inside the `if`

## Reviewers
Minimal. Feel free to remove yourself if we already have an approval from content, front-end, and UX

If you approve this one, can you also check https://github.com/fecgov/fec-eregs/pull/573 ? (As of 2021-03-02 09:00, the shared line of code is identical)

**Content** and **UX**: can you sign off on the look of terms/glossary/definitions/pop-up?

## Screenshots

Homepage News and announcements (appearance, not content)
![image](https://user-images.githubusercontent.com/26720877/109567395-46c54480-7ab3-11eb-89e2-083a093f88ff.png)

Advisory opinions, empty
![image](https://user-images.githubusercontent.com/26720877/109566740-4ed0b480-7ab2-11eb-8e7d-c485ca8ff8d4.png)

Advisory opinions, not empty
![image](https://user-images.githubusercontent.com/26720877/109566918-8d666f00-7ab2-11eb-8deb-c09da86a4448.png)

Terms (example of a citation under a list, current list on .gov, update in this PR)
![image](https://user-images.githubusercontent.com/26720877/109567164-e930f800-7ab2-11eb-9883-9866a7fb300c.png)

## Related PRs

None

## How to test

**Content**: screenshots should be cool for your review
**UX**: you, too
**Devs**: likely you, too, but if you'd like:

- pull the branch
- `npm i` (just in case)
- `npm run build`
- `./manage.py runserver`
- Check the layout of the News and announcements section of the [homepage](http://127.0.0.1:8000) vs [.gov](https://www.fec.gov)
- Go to legal advisories opinions and check one [with regulations](http://127.0.0.1:8000/data/legal/advisory-opinions/2020-02/) and [one without](http://127.0.0.1:8000/data/legal/advisory-opinions/2021-01/)
- [Coordinated communications](http://127.0.0.1:8000/help-candidates-and-committees/candidate-taking-receipts/coordinated-communications/) has a pop-up glossary option for this PR's change. At the end of Coordination defined, click `agents` to trigger the pop-up. This PR changes the way the "11 CRF…" text appears after the lists and before the next set of text
____
